integration/packer/alluxio-ami/packer.json

```diff
@@ -0,0 +1,168 @@
+{
+  "variables": {
+    "alluxio_download_url": "",
+    "alluxio_edition_flavor": "",
+    "alluxio_version": "",
+    "ami_suffix": "",
+    "aws_access_key": "",
+    "aws_region": "",
+    "aws_secret_key": "",
+    "ec2_keypair_name": "",
+    "ec2_keypair_pem_file": ""
+  },
+  "sensitive-variables": [
+    "aws_access_key",
+    "aws_secret_key"
+  ],
+  "builders": [
+    {
+      "type": "amazon-ebs",
+      "name": "alluxio-only-ami",
+      "access_key": "{{user `aws_access_key`}}",
+      "secret_key": "{{user `aws_secret_key`}}",
+      "region": "{{user `aws_region`}}",
+      "ami_block_device_mappings": [
+        {
+          "delete_on_termination": true,
+          "device_name": "/dev/xvda",
+          "volume_size": 16,
+          "volume_type": "gp2"
+        }
+      ],
+      "ami_description": "Alluxio {{user `alluxio_edition_flavor`}} {{user `alluxio_version`}} AMI",
+      "ami_name": "Alluxio {{user `alluxio_edition_flavor`}} {{user `alluxio_version`}} AMI{{user `ami_suffix`}}",
+      "ami_regions": "{{user `ami_dst_regions`}}",
+      "ami_users": "{{user `ami_users`}}",
+      "force_delete_snapshot": "{{user `ami_ovewrite_existing`}}",
+      "force_deregister": "{{user `ami_ovewrite_existing`}}",
+      "instance_type": "t3.medium",
+      "source_ami_filter": {
+        "filters": {
+          "name": "amzn-ami-hvm-*-x86_64-gp2",
+          "root-device-type": "ebs",
+          "virtualization-type": "hvm"
+        },
+        "most_recent": true,
+        "owners": [
+          "137112412989"
+        ]
+      },
+      "ssh_keypair_name": "{{user `ec2_keypair_name`}}",
+      "ssh_private_key_file": "{{user `ec2_keypair_pem_file`}}",
+      "ssh_username": "ec2-user",
+      "tags": {
+        "Origin": "",
```

Comment:
do we need to introduce more variables to optionally populate the tag values? i wonder if it is possible to use a map as a variable value and just say "tags": {{user `tags_map`}}

Reply:
packer doesn't allow user variables to be objects

Reply:
looks like they started introducing the terraform HCL to packer in a newer release: https://www.packer.io/guides/hcl/variables/#maps

guess we can't do this generically yet, but for now we still need to be able to set this `Origin` tag value for cleanup purposes? or do you have another way of setting the tag?

Reply:
we can have the origin tag be a user variable